mis_builder/models/mis_builder.py

```diff
@@ -284,6 +284,206 @@ class MisReport(models.Model):
+                 period_from, period_to,
+                 target_move,
+                 get_additional_move_line_filter=None,
+                 get_additional_query_filter=None,
+                 period_id=None):
+        """ Evaluate a report for a given period.
+
+        It returns a dictionary keyed on kpi.name with the following values:
+            * val: the evaluated kpi, or None if there is no data or an error
+            * val_r: the rendered kpi as a string, or #ERR, #DIV
+            * val_c: a comment (explaining the error, typically)
+            * style: the css style of the kpi
+                     (may change in the future!)
+            * prefix: a prefix to display in front of the rendered value
+            * suffix: a prefix to display after rendered value
+            * dp: the decimal precision of the kpi
+            * is_percentage: true if the kpi is of percentage type
+                             (may change in the future!)
+            * expr: the kpi expression
+            * drilldown: true if the drilldown method of
+                         mis.report.instance.period is going to do something
+                         useful in this kpi
+
+        :param lang_id: id of a res.lang object
+        :param aep: an AccountingExpressionProcessor instance created
+                    using _prepare_aep()
+        :param date_from, date_to: the starting and ending date
+        :param period_from, period_to: the starting and ending accounting
+                                       period (optional, if present must
+                                       match date_from, date_to)
+        :param target_move: all|posted
+        :param get_additional_move_line_filter: a bound method that takes
+                                                no arguments and returns
+                                                a domain compatible with
+                                                account.move.line
+        :param get_additional_query_filter: a bound method that takes a single
+                                            query argument and returns a
+                                            domain compatible with the query
+                                            underlying model
+        :param period_id: an optional opaque value that is returned as
+                          query_id field in the result (may change in the
+                          future!)
+        """
+        self.ensure_one()
+        res = {}
+
+        localdict = {
+            'registry': self.pool,
+            'sum': _sum,
+            'min': _min,
+            'max': _max,
+            'len': len,
+            'avg': _avg,
+            'AccountingNone': AccountingNone,
+        }
+
+        localdict.update(self._fetch_queries(
+            date_from, date_to, get_additional_query_filter))
+
+        if get_additional_move_line_filter:
+            additional_move_line_filter = get_additional_move_line_filter()
```

Comment:
If `get_additional_move_line_filter` is None, `additional_move_line_filter` will be _referenced before assignement_